sawmill-core/src/test/java/io/logz/sawmill/processors/DeDotProcessorTest.java

```diff
@@ -0,0 +1,128 @@
+            "      5,\n" +
+            "      6,\n" +
+            "      7,\n" +
+            "      8,\n" +
+            "      9\n" +
+            "    ],\n" +
+            "    \"friends\": [{\n" +
+            "        \"id\": 0,\n" +
+            "        \"first.name\": \"Sheena James\"\n" +
+            "      },\n" +
+            "      {\n" +
+            "        \"id\": 1,\n" +
+            "        \"first.name\": \"Castaneda Hinton\"\n" +
+            "      },\n" +
+            "      {\n" +
+            "        \"id\": 2,\n" +
+            "        \"first.name\": \"Della Curry\"\n" +
+            "      }\n" +
+            "    ],\n" +
+            "    \"greeting\": \"Hello, Robin! You have 10 unread messages.\",\n" +
+            "    \"favoriteFruit\": \"banana\"\n" +
+            "  },\n" +
+            "  \"_id\": \"5c9ce370543222f1cf6b53e0\",\n" +
+            "  \"index\": 0,\n" +
+            "  \"guid\": \"d055c342-4c62-4af3-944d-541a835f79a4\",\n" +
+            "  \"isActive\": false,\n" +
+            "  \"balance\": \"$3,331.30\",\n" +
+            "  \"picture\": \"http://placehold.it/32x32\",\n" +
+            "  \"age\": 29,\n" +
+            "  \"eyeColor\": \"blue\",\n" +
+            "  \"first.name\": {\n" +
+            "    \"first\": \"Magdalena\",\n" +
+            "    \"last\": \"Stewart\"\n" +
+            "  }\n" +
+            "}";
+
+    @Test
+    public void testDefaultSeperator() {
+
+        //Map<String, Object> config = createConfig("seperator","-");
+        AssertOnSeperator(messageExample, null);
+    }
+    @Test
+    public void testCustomSeperator() {
+
+        Map<String, Object> config = createConfig("seperator","_");
+        AssertOnSeperator(messageExample, config);
+
+        config = createConfig("seperator","#");
+        AssertOnSeperator(messageExample, config);
+    }
+
+
+    private void AssertOnSeperator(String messageExample, Map<String, Object> config) {
+        Doc doc = new Doc(JsonUtils.fromJsonString(Map.class,messageExample));
+
+        Object obk = doc.getSource().get("inner.object");
+
+        assertThat(doc.getSource().get("inner.object")).isNotNull();
+        assertThat(doc.getSource().get("first.name")).isNotNull();
+        assertThat(((Map<String,Object>) doc.getSource().get("inner.object")).get("gps.latitude")).isNotNull();
+        assertThat(((Map<String,Object>) doc.getSource().get("inner.object")).get("about.us")).isNotNull();
+
+        DeDotProcessor deDotProcessor = createProcessor(DeDotProcessor.class, config);
+
+        assertThat(deDotProcessor.process(doc).isSucceeded()).isTrue();
+
+        assertThat(doc.getSource().get("inner.object")).isNull();
+        assertThat(doc.getSource().get("first.name")).isNull();
+
+        String seperator = config!=null ? (String)config.get("seperator") : DeDotProcessor.Configuration.DEDOT_DEFAULT_VAL;
+
+        assertThat(doc.getSource().get("first"+ seperator +"name")).isNotNull();
```

Comment:
move the asserts to the tests functions and use copy paste instead of the reuse code function.